docs/source/agent-framework/historian-agents/forwarder/forward-historian.rst

```diff
@@ -4,58 +4,273 @@
+At a minimum, a forward historian's configuration should contain enough details to connect to and authenticate the
+remote destination.  For this it needs
+
+  1. the destination's :term:`VIP address` (`destination-vip`)
+  2. the public key of the destination server (`destination-serverkey`)
+
+There are two ways to provide these information
+
+Minimum configuration: Option 1
+-------------------------------
+
+Provide the needed information in the configuration file. For example
 
 .. code-block:: json
 
     {
-        "agentid": "forwarder",
-        "destination-vip": "ipc://@/home/volttron/.volttron/run/vip.socket"
+        "destination-vip": "tcp://172.18.0.4:22916"
+        "destination-serverkey": "D3tIAPOFf7wS3787FgEOLjoPfXUT9rAGpv80ryloZGE"
     }
 
-In order to send to a remote platform, you will need its :term:`VIP address` and server key.  The server key can be
-found by running:
+The destination server key can be found by running the following command on the **destination volttron instance**:
 
 .. code-block:: bash
 
     vctl auth serverkey
 
-Put the result into the following example:
 
 .. note::
 
-    The example shown uses the local IP address, the IP address for your configuration should match the intended target
+    The example above uses the local IP address, the IP address for your configuration should match the intended target
+
+An example configuration with above parameters is available at  `services/core/ForwardHistorian/config`.
+
+.. _config_option_2:
+
+Minimum configuration: Option 2
+-------------------------------
+
+If the destination volttron instance is web enabled then the forward historian can find the destination's vip address
+and public key using the destination's web discovery page. All web enabled volttron instances provide a
+**<instance's web address>/discovery/** page that provides the following server information
+    1. server key
+    2. vip address
+    3. instance name
+    4. RabbitMQ server's AMQP address (Only on RabbitMQ instances)
+    5. RabbitMQ server's CA cert (Only on RabbitMQ instances)
+
+To forward data to a web enabled volttron instance, forwarder can configured with the destination's web address
+instead of destination's vip address and public key. For example
 
 .. code-block:: json
 
     {
-        "agentid": "forwarder",
-        "destination-vip": "tcp://127.0.0.1:22916",
-        "destination-serverkey": "<SOME_KEY>"
+        "destination-address": "https://centvolttron2:8443"
     }
 
+An example configuration with above parameters is available at  `services/core/ForwardHistorian/config_web_address`.
 
-Adding the configuration option below will limit the backup cache to `n` gigabytes.  This will help keep a hard drive
-from filling up if the agent is disconnected from its target for a long time.
+Optional Configurations
+-----------------------
 
-::
+The most common use case for a forwarder is to forward data collected on a local instance to a remote historian. Due to
```

Comment:
Is there any other usecase?

Reply:
In theory I can think of other cases where the target is not a historian -- Forward analysis result to a different machine for post processing with a software that is not on the collector box? Forward simulation result to another machine that is connected to a device that can be sent a set_point instruction based on simulation result ?

Reply:
Analysis/simulation result is also data collected right? I didn't quite understand by - "use case for a forwarder is to forward data collected on a local instance". Anyway will pull it in.